From: Martin K. Petersen <martin.petersen@oracle.com>
Date: Tue, 22 Mar 2011 00:27:42 -0400
Subject: [PATCH] sd: Fail discard requests when logical block provisioning has been disabled
X-Git: 09b9cc44c942256026bf7a63fec2155b8f488899
References: FATE#311755
Patch-Mainline: 2.6.38

Ensure that we kill discard requests after logical block provisioning
has been disabled in sysfs.

Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Reported-by: Geert Uytterhoeven <geert@linux-m68k.org>
Reviewed-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/sd.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/scsi/sd.c b/drivers/scsi/sd.c
index 1553103..7a59696 100644
--- a/drivers/scsi/sd.c
+++ b/drivers/scsi/sd.c
@@ -592,6 +592,7 @@ static int scsi_setup_discard_cmnd(struct scsi_device *sdp, struct request *rq)
 		break;
 
 	default:
+		ret = BLKPREP_KILL;
 		goto out;
 	}
 
-- 
1.6.0.2

